gometalinter.json

```diff
@@ -20,6 +20,6 @@
     ".+\\.pb\\.gw\\.go",
     "mock_.+\\.go"
   ],
-  "Cyclo": 25,
+  "Cyclo": 40,
```

Comment:
The cyclo check is designed to make sure we don't have overly complex functions. 
Is it possible to break the offending function into several simpler functions?

Reply:
Not really, no.  

(I am not at all convinced that the gocyclo check is of any real use -- a simple count of `if` statements is very primitive, and so has false "positives" from entirely linear functions that have error checking.)

Reply:
(To put it another way, if [the error handling mechanism suggested for Go 2](https://go.googlesource.com/proposal/+/master/design/go2draft-error-handling.md) were present, the cyclo count for the function in question would drop from 27 to ~1 with no change in the function's behaviour.)

Reply:
Shifted to a `// nolint: gocyclo`.